chains/ethereum/chain.go

```diff
@@ -0,0 +1,316 @@
+package ethereum
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/simapp/params"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	transfertypes "github.com/cosmos/ibc-go/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/modules/core/02-client/types"
+	conntypes "github.com/cosmos/ibc-go/modules/core/03-connection/types"
+	chantypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
+	committypes "github.com/cosmos/ibc-go/modules/core/23-commitment/types"
+	ibcexported "github.com/cosmos/ibc-go/modules/core/exported"
+	"github.com/ethereum/go-ethereum/ethclient"
+
+	"github.com/hyperledger-labs/yui-ibc-solidity/pkg/contract/ibchandler"
+	"github.com/hyperledger-labs/yui-ibc-solidity/pkg/contract/ibchost"
+	"github.com/hyperledger-labs/yui-ibc-solidity/pkg/wallet"
+	"github.com/hyperledger-labs/yui-relayer/core"
+)
+
+type Chain struct {
+	config ChainConfig
+
+	pathEnd        *core.PathEnd
+	homePath       string
+	encodingConfig params.EncodingConfig
```

Comment:
It is a minor detail, but `encodingConfig` is removed from other chains and I think is not needed for `ethereum` either.

Reply:
You are right. This is fixed at https://github.com/hyperledger-labs/yui-relayer/pull/37/commits/30de154c36254732ed7d6e9193943691c4cddf73